<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>XWidget</class>
 <widget class="QWidget" name="XWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>827</width>
    <height>602</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <fontweight>Thin</fontweight>
   </font>
  </property>
  <property name="windowTitle">
   <string>XWidget</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* 设置顶部栏的样式 */
QWidget#top {
    background-color: #2c3e50; /* 深蓝色背景 */
    color: red; /*红色文字 */
    border-top-left-radius: 7px;     /* 左上角 */
    border-top-right-radius: 7px;     /* 右上角 */
}

#contentWidget {
    background-color:white;/* 白色背景 */
    border-bottom-left-radius: 7px;  /* 左下角 */
    border-bottom-right-radius: 7px; /* 右下角 */
}

/* 设置按钮的基本样式*/
QPushButton {
    background-color: #3498db; /* 蓝色背景 */
    border: none; /* 无边框 */
    color: white; /* 红色文字 */
    margin: 4px; /* 按钮外边距 */
    border-radius: 5px; /* 圆角边框 */
}

/* 设置按钮的基本样式*/
QPushButton#toggleButton {
    background-color:#F5F5F5; /* 蓝色背景 */
    border: none; /* 无边框 */
    color:black; /* 红色文字 */
    margin: 4px; /* 按钮外边距 */
    border-radius: 5px; /* 圆角边框 */
}

/* 鼠标悬停时按钮的样式 */
QPushButton:hover {
    background-color: #2980b9; /* 深一些的蓝色 */
}

/* 鼠标按下时按钮的样式 */
QPushButton:pressed {
    background-color: #1e3799; /* 更深的蓝色 */
}

/* 设置水平间距的样式 */
spacer#hSpacerLeft,
spacer#hSpacerMinAndMax,
spacer#hSpacerMaxAndNor,
spacer#hSpacerNorAndClose,
spacer#hSpacerRight {
    background-color: transparent; /* 透明背景 */
}

/* 设置垂直间距的样式 */
spacer#vSpacer {
    background-color: transparent; /* 透明背景 */
}


</string>
  </property>
  <layout class="QVBoxLayout" name="XWidgetLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="sizeConstraint">
    <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
   </property>
   <property name="leftMargin">
    <number>10</number>
   </property>
   <property name="topMargin">
    <number>10</number>
   </property>
   <property name="rightMargin">
    <number>10</number>
   </property>
   <property name="bottomMargin">
    <number>10</number>
   </property>
   <item>
    <widget class="QWidget" name="top" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QGridLayout" name="topLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="0" column="1">
       <widget class="QPushButton" name="pushButtonMin">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../resources/resources.qrc">
          <normaloff>:/icons/min.png</normaloff>:/icons/min.png</iconset>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="pushButtonRestore">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../resources/resources.qrc">
          <normaloff>:/icons/window.png</normaloff>:/icons/window.png</iconset>
        </property>
       </widget>
      </item>
      <item row="0" column="7">
       <widget class="QPushButton" name="pushButtonClose">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../resources/resources.qrc">
          <normaloff>:/icons/close.png</normaloff>:/icons/close.png</iconset>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="pushButtonMax">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../resources/resources.qrc">
          <normaloff>:/icons/max.png</normaloff>:/icons/max.png</iconset>
        </property>
       </widget>
      </item>
      <item row="0" column="8">
       <spacer name="hSpacerRight">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Policy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>5</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="4">
       <spacer name="hSpacerMaxAndNor">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Policy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>5</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0">
       <spacer name="hSpacerLeft">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Policy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>578</width>
          <height>13</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="6">
       <spacer name="hSpacerNorAndClose">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Policy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>5</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="2">
       <spacer name="hSpacerMinAndMax">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Policy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>5</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="contentWidget" native="true">
     <layout class="QGridLayout" name="contentWidgetLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="2" column="0" rowspan="2">
       <widget class="RoomMap" name="roomMap">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>3</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true">/* 背景和边框统一扁平风格 */
QGraphicsView {
    background-color: #f5f5f5;
    border: 1px solid #d0d0d0;
    border-radius: 6px;
}

/* 滚动条 - 极简扁平风格 */
QScrollBar:vertical, QScrollBar:horizontal {
    background: transparent;
    border: none;
    width: 8px;
    height: 8px;
    margin: 0px;
}

QScrollBar::handle:vertical, QScrollBar::handle:horizontal {
    background: #c0c0c0;
    border-radius: 4px;
}

QScrollBar::add-line, QScrollBar::sub-line {
    background: none;
    border: none;
    width: 0px;
    height: 0px;
}

QScrollBar::add-page, QScrollBar::sub-page {
    background: none;
}

/* 右键菜单 - 简洁白底灰边 */
QMenu {
    background-color: #ffffff;
    border: 1px solid #dcdcdc;
    padding: 4px;
    border-radius: 4px;
}

QMenu::item {
    padding: 5px 14px;
    background-color: transparent;
    color: #333;
}

QMenu::item:selected {
    background-color: #e0e0e0;
    color: #000000;
}

/* QAction 风格 */
QWidget QToolTip {
    background-color: #ffffff;
    color: #000000;
    border: 1px solid #c0c0c0;
}

/* 拖拽文件进入区域高亮边框（你可在 dragEnterEvent 中设置） */
RoomMap[dragActive=&quot;true&quot;] {
    border: 2px dashed #4A90E2;
    background-color: #eaf4ff;
}
</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1" rowspan="2">
       <layout class="QVBoxLayout" name="verticalLayout" stretch="0">
        <property name="spacing">
         <number>7</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="toggleButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RoomMap</class>
   <extends>QGraphicsView</extends>
   <header location="global">roommap.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
